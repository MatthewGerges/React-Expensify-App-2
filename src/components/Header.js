import React from 'react';
import {NavLink} from 'react-router-dom';


const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to = "/" activeClassName='is-active' exact = {true}>Dashboard</NavLink>
        <br/>
        <NavLink to = "/create" activeClassName='is-active'>Create Expense</NavLink>
        <br/>
        <NavLink to = "/help" activeClassName='is-active'>Get Help</NavLink>
    </header>
);

/*
removed link from header:
<br/>
        <NavLink to = "/edit" activeClassName='is-active'>Edit Expense</NavLink>
*/
//Header willl not have props only route components

//header just takes one element - a div
//switch goes through veery component and stops looking once it found a match 
//so if none match, 404 always matches
//exact will render only if paths exactly match not just include
//switch is basically replaced by a div

//path = url, component = what to show to screen
//div just takes 

export default Header;