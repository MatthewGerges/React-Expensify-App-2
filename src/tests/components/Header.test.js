//how to virtually render component without rendering it in  browser
//react-test-renderer - allows us to render stuff in jsx
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header'
import {shallow} from 'enzyme';

//shallow only renders given component not child components
//shallow rendering and full dom rendering
// test('should render Header correctly', () => {
//     const renderer = new ReactShallowRenderer();
//     renderer.render(<Header />)
//     expect(renderer.getRenderOutput()).toMatchSnapshot();
//     //tomatchSnapshot will always pass because it will be first time created so no comparison
//     //second time created - will fail if not same to first
//     console.log(renderer.getRenderOutput())
// })

//snapshot - allows us to check if header changes only at a particular point in time
//jest snapshot does not show anchor tag for navlink because only renders parents not children
//press u key to update snapshot - and make third navlink dissapear
//raf = request animation frame

test('should render Header correctly', () => {
    const wrapper = shallow(<Header />)
    // expect(toJson(wrapper)).toMatchSnapshot()
    //No need to write toJson - already configured in jest.config
    expect(wrapper).toMatchSnapshot()
    //toJson just extracts and renders the meaningful stuff from the wrapper
    // expect(wrapper.find('h1').length).toBe(1);
    // expect(wrapper.find('h1').text()).toBe('Expensify');
    //like document selector and jquery -selects class or type: '.button'
}) 