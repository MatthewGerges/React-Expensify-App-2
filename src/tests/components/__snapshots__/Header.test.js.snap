// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Get Help
  </NavLink>
</header>
`;
